<?xml version="1.0" encoding="utf-8"?>
<launch>
  <include file="$(find path_finder)/launch/map.launch"/>
  <arg name="global_env_pcd2_topic" value="/random_forest/all_map" />
  <arg name="map_size_x" value="50.0" />
  <arg name="map_size_y" value="50.0" />
  <arg name="map_size_z" value="1.5" />
  <arg name="origin_x" value=" -25.0" />
  <arg name="origin_y" value=" -25.0" />
  <arg name="origin_z" value=" -1.0" />
  <arg name="resolution" value="0.5" />

  <arg name="steer_length" value="60.0" />
  <arg name="search_radius" value="60.0" />
  <arg name="search_time" value="1" />
  <arg name="max_tree_node_nums" value="100000" />
  <arg name="use_informed_sampling" value="true" />
  <arg name="use_GUILD_sampling" value="false" />
  <arg name="run_rrt" value="false" />
  <arg name="run_rrt_star" value="false" />
  <arg name="run_rrt_sharp" value="false" />
  <arg name="run_brrt" value="true" />
  <arg name="run_brrt_star" value="false" />
  <arg name="run_brrt_optimize" value="true" />

  <arg name="brrt_optimize_p3" value="0.5" />
  <arg name="brrt_optimize_p2" value="0.4" />
  <arg name="brrt_optimize_p1" value="0.1" />
  <arg name="brrt_step" value="0.5" />

  <arg name="brrt_alpha" value="0.5" />
  <arg name="brrt_beta" value="0.3" />
  <arg name="brrt_gamma" value="0.2" />



  <node pkg="path_finder" type="path_finder" name="path_finder_node" output="screen">
    <remap from="/global_cloud" to="$(arg global_env_pcd2_topic)"/>

    <param name="run_rrt" value="$(arg run_rrt)" type="bool"/>
    <param name="run_rrt_star" value="$(arg run_rrt_star)" type="bool"/>
    <param name="run_rrt_sharp" value="$(arg run_rrt_sharp)" type="bool"/>
    <param name="run_brrt" value="$(arg run_brrt)" type="bool"/>
    <param name="run_brrt_star" value="$(arg run_brrt_star)" type="bool"/>
    <param name="run_brrt_optimize" value="$(arg run_brrt_optimize)" type="bool"/>

    <param name="occ_map/origin_x" value="$(arg origin_x)" type="double"/>
    <param name="occ_map/origin_y" value="$(arg origin_y)" type="double"/>
    <param name="occ_map/origin_z" value="$(arg origin_z)" type="double"/>
    <param name="occ_map/map_size_x" value="$(arg map_size_x)" type="double"/>
    <param name="occ_map/map_size_y" value="$(arg map_size_y)" type="double"/>
    <param name="occ_map/map_size_z" value="$(arg map_size_z)" type="double"/>
    <param name="occ_map/resolution" value="$(arg resolution)" type="double"/>

    <param name="RRT/steer_length" value="$(arg steer_length)" type="double"/>
    <param name="RRT/search_radius" value="$(arg search_radius)" type="double"/>
    <param name="RRT/search_time" value="$(arg search_time)" type="double"/>
    <param name="RRT/max_tree_node_nums" value="$(arg max_tree_node_nums)" type="int"/>

    <param name="RRT_Star/steer_length" value="$(arg steer_length)" type="double"/>
    <param name="RRT_Star/search_radius" value="$(arg search_radius)" type="double"/>
    <param name="RRT_Star/search_time" value="$(arg search_time)" type="double"/>
    <param name="RRT_Star/max_tree_node_nums" value="$(arg max_tree_node_nums)" type="int"/>
    <param name="RRT_Star/use_informed_sampling" value="$(arg use_informed_sampling)" type="bool"/>
    <param name="RRT_Star/use_GUILD_sampling" value="$(arg use_GUILD_sampling)" type="bool"/>

    <param name="RRT_Sharp/steer_length" value="$(arg steer_length)" type="double"/>
    <param name="RRT_Sharp/search_radius" value="$(arg search_radius)" type="double"/>
    <param name="RRT_Sharp/search_time" value="$(arg search_time)" type="double"/>
    <param name="RRT_Sharp/max_tree_node_nums" value="$(arg max_tree_node_nums)" type="int"/>
    <param name="RRT_Sharp/use_informed_sampling" value="$(arg use_informed_sampling)" type="bool"/>
    <param name="RRT_Sharp/use_GUILD_sampling" value="$(arg use_GUILD_sampling)" type="bool"/>

    <param name="BRRT/steer_length" value="$(arg steer_length)" type="double"/>
    <param name="BRRT/search_time" value="$(arg search_time)" type="double"/>
    <param name="BRRT/max_tree_node_nums" value="$(arg max_tree_node_nums)" type="int"/>
    <param name="BRRT/step" value="$(arg brrt_step)" type="double"/>
    

    <param name="BRRT_Star/steer_length" value="$(arg steer_length)" type="double"/>
    <param name="BRRT_Star/search_radius" value="$(arg search_radius)" type="double"/>
    <param name="BRRT_Star/search_time" value="$(arg search_time)" type="double"/>
    <param name="BRRT_Star/max_tree_node_nums" value="$(arg max_tree_node_nums)" type="int"/>
    <param name="BRRT_Star/use_informed_sampling" value="$(arg use_informed_sampling)" type="bool"/>


    <param name="BRRT_Optimize/p1" value="$(arg brrt_optimize_p1)" type="double"/>
    <param name="BRRT_Optimize/p2" value="$(arg brrt_optimize_p2)" type="double"/>
    <param name="BRRT_Optimize/p3" value="$(arg brrt_optimize_p3)" type="double"/>
    <param name="BRRT_Optimize/step" value="$(arg brrt_step)" type="double"/>

    <param name="BRRT_Optimize/alpha" value="$(arg brrt_alpha)" type="double"/>
    <param name="BRRT_Optimize/beta" value="$(arg brrt_beta)" type="double"/>
    <param name="BRRT_Optimize/gamma" value="$(arg brrt_gamma)" type="double"/>
  </node>

</launch>